spring-cloud-aws-dynamodb/src/main/java/io/awspring/cloud/dynamodb/TableNameProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.dynamodb;
+
+/**
+ * Simple interface which entity should implement when using {@link DynamoDBTemplate} save and update.
+ * Since {@link software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean} annotation does not support table name it is impossible to know what is a TableName for a given bean.
+ *
+ * @author Matej Nedic
+ */
+public interface TableNameProvider {
```

Comment:
What if we don't force users to make entities implement any interface but instead do something like:

```java
public interface TableNameResolver {
	String resolve(Object o);
}

// just a sample can be done better
class ClassNameTableNameResolver  implements TableNameResolver {

	@Override public String resolve(Object o) {
		return o.getClass().getSimpleName().toLowerCase(Locale.ROOT);
	}
}
```

`TableNameResolver` would be used in `DynamoDBTemplate` and users would be able to provide their custom implementation?

Reply:
Or perhaps even not resolving from an object, but from a `Class` (which would simplify `load` method in `DynamoDbOperations`).

Reply:
Agree with `Class` it is much better. People can then make their own implementation or we can accept `Map<Class,String>`
where String is `tableName`. Since `tableName` is not something that is changed it is pretty much static. Also we don't want to do full Spring Data project here so I think this option is great!